src/main/java/duke/RequestParser.java

```diff
@@ -0,0 +1,65 @@
+package duke;
+
+import duke.exceptions.EmptyDescriptionException;
+import duke.exceptions.EmptyTimeException;
+import duke.exceptions.IncompleteInformationException;
+import duke.exceptions.InvalidRequestException;
+
+public abstract class RequestParser {
+    public static final int TIME_INFO_START_INDEX = 1;
+
+    public static int getTaskIndex(String request) {
+        return Integer.parseInt(request.split(" ")[1]) - 1;
+    }
+
+    public static Task getTask(String request) throws Exception {
+        if (CommandType.isTodo(request)) {
+            return buildTodo(request.trim());
+        } else if (CommandType.isSpecialTask(request)) {
+            return buildSpecialTask(request.trim());
+        }
+        throw new InvalidRequestException();
+    }
```

Comment:
Do add javadoc comments for methods that are slightly more complicated to allow users to better understand workings of the code. Applies to other methods in your iP as well